pypdf/_writer.py

```diff
@@ -833,6 +833,23 @@ def _update_field_annotation(
                 ),
             ).get_object(),
         )
+
+        if "/Font" not in dr or not isinstance(dr["/Font"], DictionaryObject):
+            dr[NameObject("/Font")] = DictionaryObject()
+
+        font_dict = dr["/Font"]
+
+        # Check if the specific font (e.g., Helvetica) is in /Font
+        font_name = NameObject("/Helvetica")
```

Comment:
Why hardcoding Helvetica here?

Reply:
Mainly simplicity, I wanted to minimize changes so rather than dynamically setting it, I hardcoded it. I could set it dynamically based on the font in the /DA dictionary, but the font would have to be one provided by default or embedded dynamically, which requires more changes.

Helvetica is a default provided font, so hardcoding it ensures that it will be written.